### PR TITLE
Display unmapped schema names in error messages

### PR DESCRIPTION
**Pull request by chxwang**
### Description

Previously, Babelfish displays the mapped schema names in error messages. This commit imports a hook function remove_db_name_in_schema and uses it to restore thr unmapped name. Original unmapped name will be displayed with this commit.
 
### Issues Resolved

[BABEL-2961](https://jira.rds.a2z.com/browse/BABEL-2961)
 
### Check List

- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the PostgreSQL license, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/backend/parser/parse_func.c [hunk 1]
```diff
@@ -2062,6 +2062,10 @@ funcname_signature_string(const char *funcname, int nargs,
 
 	initStringInfo(&argbuf);
 
+	if (sql_dialect == SQL_DIALECT_TSQL && remove_db_name_in_schema_hook) {
+		funcname = (*remove_db_name_in_schema_hook)(funcname);
```
Comment (xhfanhe) [inline src/backend/parser/parse_func.c]: This is dealing with funcname, is schema name mapping needed here? 
I'm guessing it's something like `db_schema.func()` -> `schema.func()`, but is there always a schema involved?

Comment (chxwang) [reply src/backend/parser/parse_func.c]: No, thanks for pointing this out. I'll add a check to determine if there is schema included in funcname.

File: src/backend/parser/parse_func.c [hunk 2]
```diff
@@ -2059,9 +2059,22 @@ funcname_signature_string(const char *funcname, int nargs,
...
 
 	initStringInfo(&argbuf);
 
+	/* Check if there is schema in funcname */
+	while (ptr) {
+		if (*ptr == '.') {
+            break;
+        }
+        ptr++;
+	}
```
Comment (xhfanhe) [inline src/backend/parser/parse_func.c]: To minimize change in engine code, please move this logic into hook func. 
Also, I searched in BBF code base and found a function `split_object_name` that splits multiple-part name into an array. There should be some examples of using it in BBF repo.
In test cases, please also consider delimiters, upper/lower/mixed cases.

File: src/backend/parser/parse_relation.c [hunk 3]
```diff
@@ -1453,10 +1453,15 @@ parserOpenTable(ParseState *pstate, const RangeVar *relation, int lockmode)
...
+		{
+			const char * schema_name = relation->schemaname;
+			if (sql_dialect == SQL_DIALECT_TSQL && remove_db_name_in_schema_hook)
+				schema_name = (*remove_db_name_in_schema_hook)(schema_name);
 			ereport(ERROR,
 					(errcode(ERRCODE_UNDEFINED_TABLE),
 					 errmsg("relation \"%s.%s\" does not exist",
-							relation->schemaname, relation->relname)));
+							schema_name, relation->relname)));
+		}
```
Comment (RIC06X) [inline src/backend/parser/parse_relation.c:1468]: Here I think it is better to have a separate if stmt with erreport,  it is not recommended to to directly modify PG code.
You can do something like this: 
```
if (TSQL and hook function is defined)
  call hook function (within hook function, you raise the error)
else
 original ereport.
```
Also for the schema_name, can you simply use relation->schemaname?

File: src/backend/parser/parse_func.c (hunk 1 above)
Comment (xhfanhe) [reply src/backend/parser/parse_func.c]: Is this checked in new code?

File: src/backend/parser/parse_relation.c (hunk 3 above)
Comment (time-change) [reply src/backend/parser/parse_relation.c:1468]: +1

**Comment by RIC06X** [comment]
LGTM

File: src/backend/parser/parse_func.c (hunk 1 above)
Comment (xhfanhe) [reply src/backend/parser/parse_func.c]: Would you please check if `funcname` can possibly don’t contain schema name? This is the only place that doesn’t explicitly state that there must be a schema name.
- If it’s guaranteed to have schema name, then your current implementation is okay, may be better to mention that it’s caller’s responsibility to make sure that input object_name contains a schema name
- If it’s possible to have one-part funcname without schema, then you may want to add special handling for this scenario

**Review by xhfanhe — APPROVED** [review]
LGTM

**Pull request merged**